File: javaver/Cell.java
package javaver;

public class Cell {
    private Cell prevCell;
    private int value, x, y;
    private boolean visited;
    /**
     * 
     * @param value takes int value of value
     * @param prevCell takes Cell previous cell
     */
    public Cell(int value, Cell prevCell){
        this.value = value;
        this.prevCell = prevCell;
        this.visited = false;
    }
    /**
     * <p>
     * deault Cell constructor, creates coordinate at 0,0
     */
    public Cell(){
        this.prevCell = null;
        this.value = 0;
        this.x = 0;
        this.y = 0;
        this.visited = false;
    }
    /**
     * 
     * @param x takes int x coordinate
     * @param y takes int y coordinate
     * @param value takes int value of value
     */
    public Cell(int x, int y, int value){
        this.prevCell = null;
        this.value = 0;
        this.x = x;
        this.y = y;
        this.visited = false;
    }
    public int getX(){
        return this.x;
    }
    public int getY(){
        return this.y;
    }
    public void setX(int x){
        this.x = x;
    }
    public void setY(int y){
        this.y = y;
    }
    public void setValue(int value){
        this.value = value;
    }
    public int getValue(){
        return this.value;
    }
    public void setPrevCell(Cell prev){
        this.prevCell = prev;
    }
    public Cell getPrevCell(){
        return this.prevCell;
    }
    public void visit(){
        this.visited = true;
    }
    public boolean isVisited(){
        return this.visited;
    }
    @Override
    public String toString(){
        return this.x + "," + this.y;
    }
}

File: pythonver/Cell.py
class Cell:
    def __init__(self, x : int, y : int, value : int, prev_cell : 'Cell', visited : bool):
        self._x = x
        self._y = y
        self._value = value
        self._prev_cell = prev_cell
        self._visited = visited

    def get_x(self) -> int:
        return self._x
    
    def get_y(self) -> int:
        return self._y
    
    def get_value(self) -> int:
        return self._value
    
    def get_prev_cell(self) -> 'Cell':
        return self._prev_cell
    
    def get_visited(self) -> bool:
        return self._visited
    
    def visit(self) -> None:
        self._visited = True

    def set_value(self, value : int) -> None:
        self._value = value

    def __str__(self) -> str:
        return f"{self._x},{self._y}"
    
    def set_prev_cell(self, prev_cell : 'Cell') -> None:
        self._prev_cell = prev_cell

    def copy(self):
        if self._prev_cell:
            return Cell(self._x, self._y, self._value, self._prev_cell.copy(), self._visited)
        else:
            return Cell(self._x, self._y, self._value, None, self._visited)